backend/src/main/java/com/tcs/dhv/service/AuthService.java

```diff
@@ -0,0 +1,95 @@
+package com.tcs.dhv.service;
+
+import com.tcs.dhv.domain.dto.RegisterRequest;
+import com.tcs.dhv.domain.dto.UserDto;
+import com.tcs.dhv.domain.entity.User;
+import com.tcs.dhv.mapper.UserMapper;
+import com.tcs.dhv.repository.UserRepository;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.security.Keys;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.security.Key;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+@Service
+@RequiredArgsConstructor
+public class AuthService {
+
+    private final AuthenticationManager authenticationManager;
+    private final UserDetailsService userDetailsService;
+    private final PasswordEncoder passwordEncoder;
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+
+    @Value("${jwt.secret}")
+    private String secretKey;
+
+    private final Long jwtExpiryMs = 86400000L;
+
+    public UserDetails authenticate(String email, String password) {
+        authenticationManager.authenticate(
+                new UsernamePasswordAuthenticationToken(email, password)
+        );
+        return userDetailsService.loadUserByUsername(email);
+    }
+
+    public String generateToken(UserDetails userDetails) {
+        Map<String, Object> claims = new HashMap<>();
+        return Jwts.builder()
+                .setClaims(claims)
+                .setSubject(userDetails.getUsername())
+                .setIssuedAt(new Date(System.currentTimeMillis()))
+                .setExpiration(new Date(System.currentTimeMillis() + jwtExpiryMs))
+                .signWith(getSigningKey(), SignatureAlgorithm.HS256)
+                .compact();
+    }
+
+    public UserDetails validateToken(String token) {
+        String username = extractUsername(token);
+        return userDetailsService.loadUserByUsername(username);
+    }
+
+    public UserDto registerUser(RegisterRequest  request) {
+        if (userRepository.existsByEmail(request.getEmail())) {
+            throw new UsernameNotFoundException("User with this email already exists");
```

Comment:
```suggestion
            throw new IllegalArgumentException("User with this email already exists");
```

throwing `UsernameNotFound` is confusing